src/physics/land/vegetation.jl

```diff
@@ -49,7 +49,9 @@ function soil_moisture_availability!(
     r = 1/(D_top*W_cap + D_root*(W_cap - W_wilt))
 
     @inbounds for ij in eachindex(soil_moisture_availability)
-        soil_moisture_availability[ij] = r*D_top*soil_moisture[ij, 1]
+        if isfinite(soil_moisture_availability[ij])
+            soil_moisture_availability[ij] = r*D_top*soil_moisture[ij, 1]
+        end 
```

Comment:
The `isfinite` check should be on `soil_moisture` not `soil_moisture_availability` given that `soil_moisture` is overwritten anyway!